src/servers/src/error.rs

```diff
@@ -772,9 +772,14 @@ impl IntoResponse for Error {
     }
 }
 
+/// Converts [StatusCode] to [HttpStatusCode].
 pub fn status_code_to_http_status(status_code: &StatusCode) -> HttpStatusCode {
     match status_code {
-        StatusCode::Success | StatusCode::Cancelled => HttpStatusCode::OK,
+        StatusCode::Success => HttpStatusCode::OK,
+
+        // The cancelled may returned via internal client,
+        // the outside client can treat it as a gateway timeout.
+        StatusCode::Cancelled => HttpStatusCode::GATEWAY_TIMEOUT,
```

Comment:
@sunng87 @shuiyisong PTAL

Reply:
I think GATEWAY_TIMEOUT is better in clarity. 👍